### PR TITLE
feat: add utils/addInitContainerCommand

### PR DESCRIPTION
**Pull request by revolunet**
Can be used like this : 

```js
if (env.env === "dev") {
  const deployment = getDeployment(manifests);
  addInitContainerCommand(deployment, {
    command: ["yarn"],
    args: ["initdb"]
  });
}
```

### CODE REVIEW

File: src/utils/addInitContainerCommand.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ok(deployment.spec);
+  ok(deployment.spec.template);
+  ok(deployment.spec.template.spec);
+  const image = deployment.spec.template.spec.containers[0].image;
+  const initContainer = new Container({
+    args: [],
+    command: ["hostname"],
+    image,
+    imagePullPolicy: "Always",
+    name: "init",
```
Comment (douglasduteil) [inline src/utils/addInitContainerCommand.ts:19]: we might want to use the command first element as name by default for clarification

**Review by douglasduteil — APPROVED** [review]
You might want to add a failing and a "default" test case

File: src/utils/addInitContainerCommand.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  deployment: Deployment,
+  container: Partial<Container>
+) => {
+  ok(deployment.spec);
+  ok(deployment.spec.template);
+  ok(deployment.spec.template.spec);
+  const image = deployment.spec.template.spec.containers[0].image;
+  const initContainer = new Container({
+    args: [],
+    command: ["hostname"],
```
[the remark this answers is not in the document]
Comment (revolunet) [reply src/utils/addInitContainerCommand.ts]: other idea ?

File: src/utils/addInitContainerCommand.ts (hunk 1 above)
Comment (revolunet) [reply src/utils/addInitContainerCommand.ts]: `container.command[0]` ?

**Comment by revolunet** [comment]
what would be the failing test case ?

File: src/utils/addInitContainerCommand.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (douglasduteil) [reply src/utils/addInitContainerCommand.ts]: Use an assert to forcing the user the provide a command as we do not know the intent behind the init container

File: src/utils/addInitContainerCommand.ts (hunk 1 above)
Comment (douglasduteil) [reply src/utils/addInitContainerCommand.ts]: yes

**Comment by douglasduteil** [comment]
@revolunet 

> what would be the failing test case ?

A "mal formated" deployment. You can test your asserts

File: src/utils/addInitContainerCommand.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (revolunet) [reply src/utils/addInitContainerCommand.ts]: yes :+1:

**Pull request merged**